### PR TITLE
port weighted degree position string

### PR DESCRIPTION
**Pull request by LiuYuHui**

### CODE REVIEW

File: examples/meta/src/kernel/weighted_degree_position_string.sg.in [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#![create_features]
+Features feats_train = create_char_string_features(traindat, enum EAlphabet.DNA)
+Features feats_test = create_char_string_features(testdat, enum EAlphabet.DNA)
+#![create_features]
+
+#![create_kernel]
+int degree = 20
+WeightedDegreePositionStringKernel kernel(feats_train, feats_train, degree)
+kernel.init(feats_train, feats_train)
+IntVector shifts([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
```
Comment (karlnapf) [inline examples/meta/src/kernel/weighted_degree_position_string.sg.in:13]: a few questions:

* do we really need 20 here?
* what about having a few different shifts/weights here? Increases coverage maybe?

Comment (gf712) [reply examples/meta/src/kernel/weighted_degree_position_string.sg.in:13]: I guess you need as many values as vectors in feature? I am not sure how this is meant to be used though..?

Comment (karlnapf) [reply examples/meta/src/kernel/weighted_degree_position_string.sg.in:13]: I mean we can leave this for now?

Comment (gf712) [reply examples/meta/src/kernel/weighted_degree_position_string.sg.in:13]: yes, works with me

**Comment by karlnapf** [comment]
ok let's merge but this needs a rebase for data

**Pull request merged**